tests/versioned/koa-router.tap.js

```diff
@@ -0,0 +1,369 @@
+    app.use(router.routes())
+    app.use(router2.routes())
+    helper.agent.on('transactionFinished', function(tx) {
+      // NOTE: due to an implementation detail in koa-compose,
+      // sequential middleware will show up as nested. This is due to
+      // the dispatch function blocking its returned promise on the
+      // resolution of a recursively returned promise.
+      // https://github.com/koajs/compose/blob/e754ca3c13e9248b3f453d98ea0b618e09578e2d/index.js#L42-L44
+      t.exactSegments(
+        tx.trace.root, [{
+          name: 'WebTransaction/WebFrameworkUri/Koa/GET//:second',
+          children: [{
+            name: 'Koa/Router: /',
+            children: [{
+              name: 'Nodejs/Middleware/Koa/firstMiddleware//:first'
+            }, {
+              name: 'Koa/Router: /',
+              children: [{
+                name: 'Nodejs/Middleware/Koa/secondMiddleware//:second'
+              }]
+            }]
+          }]
+        }],
+        'should have expected segments'
+      )
+      t.equal(
+        tx.name,
+        'WebTransaction/WebFrameworkUri/Koa/GET//:second',
+        'transaction should be named after the middleware responsible for responding'
+      )
+      t.end()
+    })
+    run()
+  })
+
+  t.test('should work with nested routers', function(t) {
+    var Router = require('koa-router')
+    var router2 = new Router()
+    router2.get('/:second', function secondMiddleware(ctx) {
+      ctx.body = ' second'
+    })
+    router.use('/:first', router2.routes())
+    app.use(router.routes())
+    helper.agent.on('transactionFinished', function(tx) {
+      t.exactSegments(
+        tx.trace.root, [{
+          name: 'WebTransaction/WebFrameworkUri/Koa/GET//:first/:second',
+          children: [{
+            name: 'Koa/Router: /',
+            children: [{
+              name: 'Nodejs/Middleware/Koa/secondMiddleware//:first/:second'
+            }]
+          }]
+        }],
+        'should have expected segments'
+      )
+      t.equal(
+        tx.name,
+        'WebTransaction/WebFrameworkUri/Koa/GET//:first/:second',
+        'transaction should be named after the middleware responsible for responding'
+      )
+      t.end()
+    })
+    run('/123/456/')
+  })
+
+  t.test('should work with an array of paths', function(t) {
+    router.use(['/:first', '/:second'], function secondMiddleware(ctx, next) {
+      ctx.body += ' second'
+      next()
+    })
+    router.get('/:second', function terminalMiddleware(ctx) {
+      ctx.body = ' second'
+    })
+    app.use(router.routes())
+    helper.agent.on('transactionFinished', function(tx) {
+      t.exactSegments(
+        tx.trace.root, [{
+          name: 'WebTransaction/WebFrameworkUri/Koa/GET//:second',
+          children: [{
+            name: 'Koa/Router: /',
+            children: [{
+              name: 'Nodejs/Middleware/Koa/secondMiddleware//:first'
+            }, {
+              name: 'Nodejs/Middleware/Koa/secondMiddleware//:second'
```

Comment:
because the route requested matches both routes the middleware is mounted on the middleware will be fired twice.  using an array of paths will just register the same function under both routes.  i think the case in the test isn't the way the functionality is intended to be used, but it shows that it still functions as expected.

Reply:
Ah, I suppose that makes sense. Could you document that fact in the test to avoid future confusion?